File: 0033/33.c
int searchInRotatedSortedArray(struct TreeNode* root) {
    if (!root) return 0;
    int l = searchInRotatedSortedArray(root->left);
    int r = searchInRotatedSortedArray(root->right);
    return 1 + (l > r ? l : r);
}

File: 0033/33.go
func searchInRotatedSortedArray(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := searchInRotatedSortedArray(root.Left)
	r := searchInRotatedSortedArray(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}